opm/material/eos/PengRobinson.hpp

```diff
@@ -290,8 +290,8 @@ class PengRobinson
         findCriticalPoint_(Tcrit,
                            pcrit,
                            Vcrit,
-                           params.a(phaseIdx),
-                           params.b(phaseIdx));
+                           Evaluation(params.a(phaseIdx)),
+                           Evaluation(params.b(phaseIdx)) );
```

Comment:
Is there a reason for why you have to resort to an explicit conversion here?

Reply:
Basically the compilation error is that there is some deduction failure.  (I commented out the explicit Evaluation conversion.)

```
/home/kaib/OPM-devel3/opm/opm-common/opm/material/eos/PengRobinson.hpp:290:27: error: no matching function for call to ‘Opm::PengRobinson<float, false>::findCriticalPoint_(Opm::DenseAd::Evaluation<float, 3>&, Opm::DenseAd::Evaluation<float, 3>&, Opm::DenseAd::Evaluation<float, 3>&, float, float)’
  290 |         findCriticalPoint_(Tcrit,
      |         ~~~~~~~~~~~~~~~~~~^~~~~~~
  291 |                            pcrit,
      |                            ~~~~~~
  292 |                            Vcrit,
      |                            ~~~~~~
  293 |                            /*Evaluation */(params.a(phaseIdx)),
      |                            ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
  294 |                            /*Evaluation */(params.b(phaseIdx)) );
      |                            ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/home/kaib/OPM-devel3/opm/opm-common/opm/material/eos/PengRobinson.hpp:306:17: note: candidate: ‘template<class Evaluation> static void Opm::PengRobinson<Scalar, UseLegacy>::findCriticalPoint_(Evaluation&, Evaluation&, Evaluation&, const Evaluation&, const Evaluation&) [with Evaluation = Evaluation; Scalar = float; bool UseLegacy = false]’
  306 |     static void findCriticalPoint_(Evaluation& Tcrit,
      |                 ^~~~~~~~~~~~~~~~~~
/home/kaib/OPM-devel3/opm/opm-common/opm/material/eos/PengRobinson.hpp:306:17: note:   template argument deduction/substitution failed:
/home/kaib/OPM-devel3/opm/opm-common/opm/material/eos/PengRobinson.hpp:290:27: note:   deduced conflicting types for parameter ‘const Evaluation’ (‘Opm::DenseAd::Evaluation<float, 3>’ and ‘float’)
  290 |         findCriticalPoint_(Tcrit,
      |         ~~~~~~~~~~~~~~~~~~^~~~~~~
  291 |                            pcrit,
      |                            ~~~~~~
  292 |                            Vcrit,
      |                            ~~~~~~
  293 |                            /*Evaluation */(params.a(phaseIdx)),
      |                            ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
  294 |                            /*Evaluation */(params.b(phaseIdx)) );
      |                            ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
```

Reply:
>> Is there a reason for why you have to resort to an explicit conversion here?
>
> Basically the compilation error is that there is some deduction failure.

Ah, yeah, right.  That makes sense.  There are ways to exclude the `a` and `b` parameters from template argument deduction in the function definition for `findCriticalPoint_()`, but none of those are particularly easy to read.